### DISCUSSION TITLE
How to contribute in open-source?

### DISCUSSION THREAD

**Discussion by youssef-abbih** [fastapi/fastapi#4713] [Questions] [answered]
hello,
I know this is not the right place to ask this question,   I just want to know how as a beginner can I contribute to an open source project I choose FastAPI because I used it a lot in my personal projects .

Thanks.

**Comment by lnxpy** [comment, marked as answer]
Hi Youssef,

Always start small. Try contributing to the small packages and projects. Read their issues and find solutions for them. You said you have being using this framework for a while. That's awesome but it just gives you a POV about how this framework actually works and a general view about its different parts and processes. Since FastAPI is written in Python, you need to know the different Pythonic conventions and best practices in order to either refactor a part from it or add a new feature that you think would be perfect in this project.

To start your journey, I suggest you contributing to the documentations and improving the test cases of the project. :)
